app/jobs/regular/grant_badge.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+module Jobs
+  class GrantBadge < ::Jobs::Base
+    def execute(args)
+      return unless SiteSetting.enable_badges
+
+      badge = Badge.enabled.find_by(id: args[:badge_id])
+      return unless badge
+
+      # Cancels the scheduled job to ensure badge consistency as the badges are
```

Comment:
I don't think this is sufficient since Sidekiq jobs can be processed concurrently. The `Jobs::EnsureBadgeConsistency` job could already be running. If you want to ensure that stuff in the `ensure_badge_consistency` job is not running while this job is running, you will have to use some sort of mutex.

Reply:
O I see a distributed mutex is being used below.